src/mixxx.cpp

```diff
@@ -1129,20 +1132,15 @@ bool MixxxMainWindow::event(QEvent* e) {
 void MixxxMainWindow::closeEvent(QCloseEvent *event) {
     if (!confirmExit()) {
         event->ignore();
+        return;
     }
+    // Save the current window state (position, maximized, etc)
```

Comment:
Most shutdown code is in MixxxMainWindow::finalize -- this should probably go there.